docs/DevGuide/Protocols.md

```diff
@@ -48,26 +64,40 @@ has explicitly defined (and documented!) the interface they expect. And the
 developer who consumes the protocol by writing classes that conform to it knows
 exactly what needs to be implemented.
 
+Note that the `conforms_to` metafunction is SFINAE-friendly, so you can also
+use it like this:
+
+\snippet Utilities/Test_ProtocolHelpers.cpp protocol_sfinae
+
 We typically define protocols in a file named `Protocols.hpp` and within a
 `protocols` namespace, similar to how we write \ref DataBoxTagsGroup "tags" in a
 `Tags.hpp` file and within a `Tags` namespace. The file should be placed in the
-in the directory associated with the code that depends on classes conforming to
-the protocols. For example, the protocol `Named` in the example above would be
+directory associated with the code that depends on classes conforming to the
+protocols. For example, the protocol `Named` in the example above would be
 placed in directory that also has the `greet` function.
 
-## Protocol users: Testing protocol conformance
+# Protocol users: Testing protocol conformance {#protocols_testing_conformance}
 
 Any class that indicates it conforms to the protocol must test that it actually
 does using the `test_protocol_conformance` metafunction from
 `tests/Unit/ProtocolTestHelpers.hpp`:
 
 \snippet Utilities/Test_ProtocolHelpers.cpp test_protocol_conformance
 
-## Protocol authors: Protocols should not be templates
+# Protocol authors: Protocols must be unary type traits {#protocols_author}
```

Comment:
I'm not sure I completely agree with the suggestions in this section anymore now that the protocols are unary type traits. If we want to pass an extra template parameter, we could for instance define a struct with a metafunction within:
```
template <typename NameType>
struct is_name_callable {
template <typename ToCheck>
using check_conformance = std::conditional_t< (checks) >;
};
```
Then the inheritance can be `ConformsTo<is_name_callable<ThisNameType>::check_conformance>` and similar in the conformance checking metafunction. This gives the same flexibility as extra template parameters without disrupting the interface you need to supply to the checking metafunctions. What do you think?

Reply:
- In your example the `check_conformance` is the protocol, so it's still a unary type trait.
- I think the statement still holds that all potential template parameters for a protocol must be known to the class that conforms to it (via the `ConformsTo` inheritance) and therefore should be made part of the protocol. In your example it's `ThisNameType`. Instead of passing `ThisNameType` to the struct that "generates" the `check_conformance` metafunction, you can just make `NameType` part of the interface (as described in the docs).
- This guideline also encourages classes to publicly expose their template parameters as type aliases, which is a good thing to do. There's no reason that I can think of why the class in you example shouldn't expose its `ThisNameType` template parameter as a type alias.

Reply:
Yeah, alright. I'm convinced that this is a fine strategy for almost all use cases. It's probably also true that anyone who finds a complex enough edge-case to want the nested type templates will also be able to figure that solution out for themselves.